app/models/manageiq/providers/workflows/automation_manager/configuration_script_source.rb

```diff
@@ -33,4 +41,17 @@ def sync
     update!(:status => "error", :last_updated_on => Time.zone.now, :last_update_error => error)
     raise error
   end
+
+  def each_file
+    git_repository.update_repo
+    git_repository.with_worktree do |worktree|
+      worktree.ref = scm_branch
+      worktree.blob_list.each do |filename|
+        next if filename.start_with?(".") || !filename.end_with?(".asl")
+
+        payload = worktree.read_file(filename)
+        yield filename, payload
+      end
+    end
+  end
```

Comment:
Think this would be better in core's EmbeddedAutomationManager repository class but I wanted to test it out self-contained here.